pkg/webhook/validation/dynakube/oneagent.go

```diff
@@ -106,36 +107,51 @@ func hasConflictingMatchLabels(labelMap, otherLabelMap map[string]string) bool {
 	return labelSelector.Matches(otherLabelSelectorLabels) || otherLabelSelector.Matches(labelSelectorLabels)
 }
 
-func hasOneAgentVolumeStorageEnabled(dynakube *dynatracev1beta2.DynaKube) (isEnabled bool, isSet bool) {
-	envVar := env.FindEnvVar(dynakube.GetOneAgentEnvironment(), oneagentEnableVolumeStorageEnvVarName)
+func hasOneAgentVolumeStorageEnabled(dk *dynakube.DynaKube) (isEnabled bool, isSet bool) {
+	envVar := env.FindEnvVar(dk.GetOneAgentEnvironment(), oneagentEnableVolumeStorageEnvVarName)
 	isSet = envVar != nil
 	isEnabled = isSet && envVar.Value == "true"
 
 	return
 }
 
-func unsupportedOneAgentImage(_ context.Context, _ *dynakubeValidator, dynakube *dynatracev1beta2.DynaKube) string {
-	if env.FindEnvVar(dynakube.GetOneAgentEnvironment(), oneagentInstallerScriptUrlEnvVarName) != nil ||
-		env.FindEnvVar(dynakube.GetOneAgentEnvironment(), oneagentInstallerTokenEnvVarName) != nil {
+func unsupportedOneAgentImage(_ context.Context, _ *dynakubeValidator, dk *dynakube.DynaKube) string {
+	if env.FindEnvVar(dk.GetOneAgentEnvironment(), oneagentInstallerScriptUrlEnvVarName) != nil ||
+		env.FindEnvVar(dk.GetOneAgentEnvironment(), oneagentInstallerTokenEnvVarName) != nil {
 		return warningOneAgentInstallerEnvVars
 	}
 
 	return ""
 }
 
-func conflictingOneAgentVolumeStorageSettings(_ context.Context, _ *dynakubeValidator, dynakube *dynatracev1beta2.DynaKube) string {
-	volumeStorageEnabled, volumeStorageSet := hasOneAgentVolumeStorageEnabled(dynakube)
-	if dynakube.NeedsReadOnlyOneAgents() && volumeStorageSet && !volumeStorageEnabled {
+func conflictingOneAgentVolumeStorageSettings(_ context.Context, _ *dynakubeValidator, dk *dynakube.DynaKube) string {
+	volumeStorageEnabled, volumeStorageSet := hasOneAgentVolumeStorageEnabled(dk)
+	if dk.NeedsReadOnlyOneAgents() && volumeStorageSet && !volumeStorageEnabled {
 		return errorVolumeStorageReadOnlyModeConflict
 	}
 
 	return ""
 }
 
-func conflictingHostGroupSettings(_ context.Context, _ *dynakubeValidator, dynakube *dynatracev1beta2.DynaKube) string {
-	if dynakube.HostGroupAsParam() != "" {
+func conflictingHostGroupSettings(_ context.Context, _ *dynakubeValidator, dk *dynakube.DynaKube) string {
+	if dk.HostGroupAsParam() != "" {
 		return warningHostGroupConflict
 	}
 
 	return ""
 }
+
+var oneAgentSemVerVersionSpec = regexp.MustCompile(`^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)$`)
```

Comment:
do we need this?

the `semver` package has a `IsValid` func.
the only downside of it is that `semver` always has a `v` prefix.

we already have some implementation that tries to handle this here: https://github.com/Dynatrace/dynatrace-operator/blob/5aa5128aa7fc057747cbae8edbdf9b5404f8e942/pkg/util/dtversion/dtversion.go#L26

Reply:
unfortunately we do in my opinion.

reason: we want to allow only a subset of SemVer: major.minor.patch without prerelease or build parts that would be allowed due to SemVer spec.

To technically speaking it is not SemVer at all 😄

Reply:
changed to Adams suggestion in 3d3c8add5a6a1bb9ef396db5a088d07373068bbe